frontend/server/controllers/ContestController.php

```diff
@@ -336,92 +336,109 @@ private static function isInvitedToContest(
-                    $needsInformation,
-                    $requestUserInformation
-                ] = ContestsDAO::getNeedsInformation($contest->problemset_id);
-                $identity = $session['identity'];
-                $result['smartyProperties']['needsBasicInformation'] = $needsInformation && (
+            }
+
+            [
+                'needsBasicInformation' => $result['needsBasicInformation'],
+                'requestsUserInformation' => $result['requestsUserInformation'],
+            ] = ContestsDAO::getNeedsInformation($contest->problemset_id);
+            $identity = $session['identity'];
+
+            $result['needsBasicInformation'] =
+                $result['needsBasicInformation'] && (
                     !$identity->country_id || !$identity->state_id ||
                     !$identity->school_id
-                );
-                // Privacy Statement Information
-                $privacyStatementMarkdown = PrivacyStatement::getForProblemset(
-                    $identity->language_id,
-                    'contest',
-                    $requestUserInformation
-                );
-                if (!is_null($privacyStatementMarkdown)) {
-                    $statementType = "contest_{$requestUserInformation}_consent";
-                    $result['smartyProperties']['privacyStatement'] = [
-                        'markdown' => $privacyStatementMarkdown,
-                        'gitObjectId' =>
-                            PrivacyStatementsDAO::getLatestPublishedStatement(
-                                $statementType
-                            )['git_object_id'],
-                        'statementType' => $statementType
-                    ];
-                }
-                $result['smartyProperties']['requestsUserInformation'] = $requestUserInformation;
+            );
+
+            // Privacy Statement Information
+            $privacyStatementMarkdown = PrivacyStatement::getForProblemset(
+                $identity->language_id,
+                'contest',
+                $result['requestsUserInformation']
+            );
+            if (!is_null($privacyStatementMarkdown)) {
+                $statementType =
+                    "contest_{$result['requestsUserInformation']}_consent";
+                $result['privacyStatement'] = [
+                    'markdown' => $privacyStatementMarkdown,
+                    'gitObjectId' =>
+                        PrivacyStatementsDAO::getLatestPublishedStatement(
+                            $statementType
+                        )['git_object_id'],
+                    'statementType' => $statementType
+                ];
+            }
+        } catch (ApiException $e) {
+            throw $e;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Show the contest intro unless you are admin, or you already started this
+     * contest.
+     * @param Request $r
+     * @return bool
+     */
+    public static function showIntro(
```

Comment:
```suggestion
    public static function shouldShowIntro(
```

`showIntro` suena a que esto va a desplegar la introducción, y en realidad lo que está haciendo es determinando si se debería desplegar.

Reply:
Listo